src/OrchardCore.Modules/OrchardCore.Workflows/Services/WorkflowManager.cs

```diff
@@ -105,57 +113,79 @@ public Task<ActivityContext> CreateActivityExecutionContextAsync(ActivityRecord
         public async Task TriggerEventAsync(string name, IDictionary<string, object> input = null, string correlationId = null, bool isExclusive = false, bool isAlwaysCorrelated = false)
         {
             var activity = _activityLibrary.GetActivityByName(name);
-
             if (activity == null)
             {
                 _logger.LogError("Activity '{ActivityName}' was not found", name);
                 return;
             }
 
-            // Look for workflow types with a corresponding starting activity.
-            var workflowTypesToStart = await _workflowTypeStore.GetByStartActivityAsync(name);
-
-            // And any workflow halted on this kind of activity for the specified target.
+            // Resume workflow instances halted on this kind of activity for the specified target.
             var haltedWorkflows = await _workflowStore.ListByActivityNameAsync(name, correlationId, isAlwaysCorrelated);
-
-            // If no workflow matches the event, do nothing.
-            if (!workflowTypesToStart.Any() && !haltedWorkflows.Any())
+            foreach (var workflow in haltedWorkflows)
             {
-                return;
+                // Don't allow scope recursion per workflow instance id.
+                if (_recursions.TryGetValue(workflow.WorkflowId, out var count) && count > 0)
+                {
+                    continue;
```

Comment:
Should we log some of this? Even at `Information` or `Debug` as you prefer?

Reply:
Yes, i hesitated, a WF listens to events that may be triggered by other WFs (through handlers), but in a given scope, it is not listening to its own events by design, like a message bus that just ignore a message sent by yourself.

Anyway, i will add it as it may be useful